packages/react/src/components/RadioButton/next/RadioButton.js

```diff
@@ -0,0 +1,126 @@
+import PropTypes from 'prop-types';
+import React, { useRef } from 'react';
+import classNames from 'classnames';
+import { usePrefix } from '../../../internal/usePrefix';
+import getUniqueId from '../../../tools/uniqueId';
+import { Text } from '../../Text';
+
+const RadioButton = React.forwardRef(function RadioButton(
+  {
+    className,
+    hideLabel,
+    id,
+    labelPosition = 'right',
+    labelText = '',
+    name,
+    onChange = () => {},
+    value = '',
+    ...other
+  },
+  ref
+) {
+  const prefix = usePrefix();
+  const { current: uid } = useRef(getUniqueId());
+  const uniqueId = id || uid;
+
+  const handleChange = (evt) => {
+    onChange(value, name, evt);
+  };
```

Comment:
Style nit: use function declarations versus arrow functions

```suggestion
  function handleOnChange(event) {
    onChange(value, name, event);
  }
```